src/meson.build

```diff
@@ -291,26 +291,22 @@ hb_gobject_headers = files(
   'hb-gobject-structs.h',
 )
 
-ragel = find_program('ragel', version: '6.10', required: false)
+ragel = find_program('ragel', version: '6.10', required: get_option('ragel'))
 has_ragel = ragel.found()
-if not has_ragel and get_option('ragel_subproject')
-    ragel = subproject('ragel').get_variable('ragel')
-    has_ragel = true
-endif
 if not has_ragel
-  warning('You have to install ragel if you are going to develop HarfBuzz itself')
-else
-  ragel_helper = find_program('gen-ragel-artifacts.py')
-  foreach rl : hb_base_ragel_sources
-    hh = rl.split('.')[0] + '.hh'
-    custom_target('@0@'.format(hh),
-      build_by_default: true,
-      input: rl,
-      output: hh,
-      command: [ragel_helper, ragel, '@OUTPUT@', meson.current_source_dir(), '@INPUT@'],
-    )
-  endforeach
+  message('You have to install ragel if you are going to develop HarfBuzz itself')
+  ragel = 'error'
 endif
+ragel_helper = find_program('gen-ragel-artifacts.py')
+foreach rl : hb_base_ragel_sources
```

Comment:
I think this part should remain in the `else` case of `if not has_ragel`, that's what make the CI fail AFAIK.